src/databricks/labs/remorph/reconcile/connectors/snowflake.py

```diff
@@ -1,16 +1,50 @@
 from pyspark.sql import DataFrame
 
 from databricks.labs.remorph.reconcile.connectors.data_source import DataSource
+from databricks.labs.remorph.reconcile.constants import SourceDriver
 from databricks.labs.remorph.reconcile.recon_config import Schema, Tables
 
 
 class SnowflakeDataSource(DataSource):
+
+    @property
+    def get_jdbc_url(self) -> str:
+        url = f"jdbc:{self.source}://{self.connection_params['account']}.snowflakecomputing.com/?user={self.connection_params['sfUser']}&password={self.connection_params['sfPassword']}&db={self.connection_params['sfDatabase']}&schema={self.connection_params['sfSchema']}&warehouse={self.connection_params['sfWarehouse']}"
+        if 'sfRole' in self.connection_params:
+            url = url + f"&role={self.connection_params['sfRole']}"
+        return url
+
     def read_data(self, schema_name: str, catalog_name: str, table_or_query: str, table_conf: Tables) -> DataFrame:
-        # Implement Snowflake-specific logic here
-        return NotImplemented
+        if table_conf.jdbc_reader_options is None:
+            df = self.reader(table_or_query)
+        else:
```

Comment:
simplify remove else loop and add a negative check first.

Reply:
Yes. Simplified it. Done